Simbody/tests/TestParallelForces.cpp

```diff
@@ -0,0 +1,168 @@
+/* -------------------------------------------------------------------------- *
+ *                               Simbody(tm)                                  *
+ * -------------------------------------------------------------------------- *
+ * This is part of the SimTK biosimulation toolkit originating from           *
+ * Simbios, the NIH National Center for Physics-Based Simulation of           *
+ * Biological Structures at Stanford, funded under the NIH Roadmap for        *
+ * Medical Research, grant U54 GM072970. See https://simtk.org/home/simbody.  *
+ *                                                                            *
+ * Portions copyright (c) 2007-15 Stanford University and the Authors.        *
+ * Authors: Thomas Lau                                                    *
+ * Contributors:                                                              *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include "SimTKsimbody.h"
+#include <time.h>
+#include <chrono>
+#include <thread>
+#include <iostream>
+
+using namespace SimTK;
+using namespace std;
+
+class ParallelForceImpl : public Force::Custom::Implementation {
+public:
+    mutable bool hasRealized[Stage::Report+1];
+    ParallelForceImpl() {
+        for (int i = 0; i < Stage::NValid; i++)
+            hasRealized[i] = false;
+    }
+    bool shouldBeParallelIfPossible() const override{
+      return true;
+    }
+    void calcForce(const State& state, Vector_<SpatialVec>& bodyForces, Vector_<Vec3>& particleForces, Vector& mobilityForces) const {
+         std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+    Real calcPotentialEnergy(const State& state) const {
+        return 0.0;
+    }
+    void realizeTopology(State& state) const {
```

Comment:
I think these are virtual method overrides aren't they? If so they should say `override`.